### ISSUE TITLE
Background Process for Auto Syncing Git Repos was skipped

### ISSUE THREAD

**Issue by linnabraham** [GitJournal/git-auto-sync#9]
Hi Vishesh,
Thanks for putting your efforts into this. I was making use of a bash script and cron until now. I was thinking of moving over to inode watcher tools like `entr` or something. So I was excited to see that you were focusing on this feature.

However, this doesnt seem to work for me on ArchLinux. The manual mode seemed to create commits but didnt push to the remote. I saw this error.

 ```gpg-agent[2775]: scdaemon[8002]: apdu_open_reader: failed to open driver 'libpcsclite.so.1': libpcsclite.so.1: cannot open shared object file: No such file or directory```

I installed the pcsclite package and this seemed to solve the manual mode issue to some extent. When i check the systemd status of the daemon I see this error. 
```
systemd[3395]: Background Process for Auto Syncing Git Repos was skipped because of a failed condition check (ConditionFileIsExecutable=/usr/bin/git-auto-sync-daemon)
```
The gpg-agent is still not completely silent. I have these errors still lurking. 
```gpg-agent[8000]: scdaemon[8000]: pcsc_establish_context failed: no service```

Can you help me troubleshoot this?

**Comment by vHanda** [comment]
Could you please make sure the git-auto-sync-daemon is executable? That's what the error says.

I'll work on adding a proper snap package so all of this is correctly handled.

**Comment by vHanda** [comment]
I realized that the release didn't actually contain the `git-auto-sync-daemon` binary. I've fixed that. Could you please try it out again?

I'm sill in the process of making a snap (pending review)

**Comment by linnabraham** [comment]
Yes. I did that and that error seems to have disappeared. However I get a new error which might be specific to my setup I am not sure.
```
May 19 12:23:52 archlinux git-auto-sync-daemon[4517]: 2022/05/19 12:23:52 exit status 128: Command: git fetch origin
May 19 12:23:52 archlinux git-auto-sync-daemon[4517]: StdOut:
May 19 12:23:52 archlinux git-auto-sync-daemon[4517]: StdErr: git@github.com: Permission denied (publickey).
May 19 12:23:52 archlinux git-auto-sync-daemon[4517]: fatal: Could not read from remote repository.
May 19 12:23:52 archlinux git-auto-sync-daemon[4517]: Please make sure you have the correct access rights
May 19 12:23:52 archlinux git-auto-sync-daemon[4517]: and the repository exists.
```
I think i have setup ssh to use one of my gpg subkeys instead of a separate ssh key and I have the following two lines in my .zshrc
```
export SSH_AUTH_SOCK=$(gpgconf --list-dirs agent-ssh-socket)
gpgconf --launch gpg-agent
```
